### ISSUE TITLE
What to do with RoomGraphTest

### ISSUE THREAD

**Issue by koosemose** [TeamPorcupine/ProjectPorcupine#1683]
Currently World.cs has the RoomGraphTest function, which simply tests that rooms generated as expected with connections as expected. However, these are hardcoded based on the current starting area, and any time things are changed about the room structure, their connections, or how the starting area is loaded, RoomGraphTest has to be tweaked, most (if not all) failures of this function I've seen have nothing to do with the RoomGraph working incorrectly, but instead are because the starting area has changed and RoomGraphTest hasn't been updated.

Should RoomGraphTest just be completely removed, changed to have a save file save some kind of error-checking code that tells RoomGraphTest what to expect, or changed to a unit test that operates off of a set save file (I'm not entirely sure how feasible this actually is as a Unit Test).

**Comment by koosemose** [comment]
It may be VERY difficult (if not quite impossible) to do it from a UnitTest. It requires creating a world (which in turn gives us a roomgraph to test), and creating a world from a unit test requires nearly the entirety of the game to be implemented in said Unit Test, to support everything that is needed to create a world, including reading in all the lua, all the data files, instantiating all the various events. 

It may be possible to do, but it's beyond me how to do it in a reasonable manner. If anyone else wants to take it up, I've created a save file with a minimalistic base (same general layout as the current starting base, but with rooms only 1 tile, just enough to test the room graph), let me know if you want to give it a try and I can give you the file, and pass on what bits I was able to figure out in case they may help.
